server/chat-plugins/room-events.ts

```diff
@@ -30,19 +53,25 @@ function formatEvent(event: RoomEvent, showAliases?: boolean) {
 function getAllAliases(room: Room) {
 	if (!room.settings.events) return [];
 	const aliases: string[] = [];
-	for (const event of Object.values(room.settings.events)) {
-		if (event.aliases) aliases.push(...event.aliases);
+	for (const aliasID of Object.keys(room.settings.events).filter(id => 'eventID' in room.settings.events![id])) {
+		aliases.push(aliasID);
 	}
 	return aliases;
 }
+
+function getAllCategories(room: Room) {
+	if (!room.settings.events) return [];
+	const categories: string[] = [];
+	for (const categoryID of Object.keys(room.settings.events).filter(id => 'events' in room.settings.events![id])) {
+		categories.push(categoryID);
+	}
+	return categories;
+}
+
 function getEventID(nameOrAlias: string, room: Room): ID {
 	let id = toID(nameOrAlias);
-	if (room.settings.events && !room.settings.events[id]) {
-		for (const possibleEvent in room.settings.events) {
-			if (room.settings.events[possibleEvent].aliases?.includes(id)) {
-				id = toID(possibleEvent);
-			}
-		}
+	if (room.settings.events?.[id] && 'eventID' in room.settings.events[id]) {
+		id = (room.settings.events[id] as RoomEventAlias).eventID;
```

Comment:
```suggestion
	const event = room.settings.events?.[id];
	if (event && 'eventID' in event) {
		id = event.eventID;
```

(It's generally at three repetitions that I start using intermediate variables.)